lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -24,6 +26,21 @@
     'lti',
 ]
 
+# exclude these in Metrics
+NON_STUDENT_ROLES = ['instructor', 'staff']
+
+
+def get_non_student_list(course_key):
+    """
+    Find all user_ids with instructor or staff roles in student_courseaccessrole table
+    """
+    non_students = CourseAccessRole.objects.select_related('user').filter(
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user__id')
+
+    return [non_student['user__id'] for non_student in non_students]
```

Comment:
That 2nd idea was a bad one:

select course_id, count(user_id), role
from student_courseaccessrole
where role in ('staff', 'instructor')
group by user_id, course_id
having count(user_id) = 1
and role = 'instructor'
order by count(user_id) desc

gives a bunch of rows where the user is an instructor but there is only 1 row.